Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitClient.swift

```diff
@@ -135,10 +153,21 @@ import BraintreeCore
         context: ASWebAuthenticationPresentationContextProviding,
         completion: @escaping (Bool, Error?) -> Void
     ) {
-        apiClient.sendAnalyticsEvent("ios.sepa-direct-debit.web-flow.started")
-        self.webAuthenticationSession.start(url: url, context: context) { url, error in
-            self.handleWebAuthenticationSessionResult(url: url, error: error, completion: completion)
-        }
+        self.webAuthenticationSession.start(
+            url: url,
+            context: context,
+            sessionDidDisplay: { [weak self] didDisplay in
+                if didDisplay {
+                    self?.apiClient.sendAnalyticsEvent(BTSEPADirectAnalytics.challengePresentationSucceeded)
+                    self?.isAuthenticationSessionStarted = true
+                } else {
+                    self?.apiClient.sendAnalyticsEvent(BTSEPADirectAnalytics.challengePresentationFailed)
```

Comment:
I think I also need to add two errors in the BTSepaDirectDebitError and add it in Changelog.